util.js

```diff
@@ -62,4 +201,171 @@ Util.prototype.kill = function(pid) {
     .exec();
 };
 
+/**
+ * Push a local file to a remote destination on device
+ * @param {String} local Path to send to device
+ * @param {String} remote Destination on device to receive files from local
+ * @returns {Promise}
+ */
+Util.prototype.push = function(local, remote) {
+  return new Command()
+    .env('ANDROID_SERIAL', this.serial)
+    .adb('push ' + local + ' ' + remote)
+    .exec();
+};
+
+/**
+ * Pull a remote file from the device to a local destination
+ * @param {String} remote The remote path to fetch from device
+ * @param {String} [local] The local destination to store files from device.
+ *                         Defaults to current working direction if not
+ *                         specified.
+ * @returns {Promise}
+ */
+Util.prototype.pull = function(remote, local) {
+  return new Command()
+    .env('ANDROID_SERIAL', this.serial)
+    .adb('pull ' + remote + ' ' + (local || process.cwd()))
+    .exec();
+};
+
+/**
+ * Fetch a remote file on device to a local temporary location. Promise resolves
+ * with temporary path where file is located.
+ * @param {String} remotePath Location of file on device
+ * @param {String} file File to pull from `remotePath`
+ * @returns {Promise}
+ */
+Util.prototype.pullToTemp = function(remotePath, file) {
+  var util = this;
+
+  return new Promise(function(resolve, reject) {
+    temp.mkdir('raptor-temp', function(err, tempDir) {
+      if (err) {
+        return reject(err);
+      }
+
+      var remote = path.join(remotePath, file);
+      var local = path.join(tempDir, file);
+
+      util
+        .pull(remote, local)
+        .then(function() {
+          fs.exists(local, function(exists) {
+            if (!exists) {
+              return reject(new Error('Failed to pull remote file'));
+            }
+
+            resolve(tempDir);
+          })
+        }, reject);
+    });
+  });
+};
+
+/**
+ * Pull the Settings application.zip from device
+ * @returns {Promise}
+ */
+Util.prototype.pullApplicationZip = function() {
+  var util = this;
+  var appZip = 'application.zip';
+
+  // application.zip could be in one of two different locations on device.
+  // Fall back to other location if first failed.
+  return new Promise(function(resolve, reject) {
+    util
+      .pullToTemp(B2G_SETTINGS_PATH, appZip)
+      .then(resolve, function() {
+        util
+          .pullToTemp(DATA_LOCAL_SETTINGS_PATH, appZip)
+          .then(resolve, reject)
+      });
+  });
+};
+
+/**
+ * Pull a B2G INI file from device
+ * @returns {Promise}
+ */
+Util.prototype.pullB2GIni = function() {
+  var util = this;
+
+  // The B2G INI file is either application.ini or platform.ini. Fall back to
+  // secondary location if first fails.
+  return new Promise(function(resolve, reject) {
+    util
+      .pullToTemp(SYSTEM_B2G, 'application.ini')
+      .then(function(tempDir) {
+        resolve(path.join(tempDir, 'application.ini'))
```

Comment:
Nit: maybe make a local var for 'application.ini' so if it changes you only need to update it in one location